src/Sylius/Bundle/CartBundle/spec/Provider/CartProviderSpec.php

```diff
@@ -30,79 +33,100 @@ function let(
         CartContextInterface $context,
         ObjectManager $manager,
         FactoryInterface $cartFactory,
-        RepositoryInterface $repository,
+        RepositoryInterface $cartRepository,
         EventDispatcherInterface $eventDispatcher
     ) {
-        $this->beConstructedWith($context, $manager, $cartFactory, $repository, $eventDispatcher);
+        $this->beConstructedWith($context, $manager, $cartFactory, $cartRepository, $eventDispatcher);
     }
 
     function it_is_initializable()
     {
-        $this->shouldHaveType('Sylius\Bundle\CartBundle\Provider\CartProvider');
+        $this->shouldHaveType(CartProvider::class);
     }
 
     function it_implements_Sylius_cart_provider_interface()
     {
-        $this->shouldImplement('Sylius\Component\Cart\Provider\CartProviderInterface');
+        $this->shouldImplement(CartProviderInterface::class);
     }
 
     function it_looks_for_cart_by_identifier_if_any_in_storage(
-        $context,
-        $repository,
-        $eventDispatcher,
+        CartContextInterface $context,
+        RepositoryInterface $cartRepository,
+        EventDispatcherInterface $eventDispatcher,
         CartInterface $cart
     ) {
         $context->getCurrentCartIdentifier()->willReturn(3);
-        $repository->find(3)->willReturn($cart);
-        $eventDispatcher->dispatch(SyliusCartEvents::CART_INITIALIZE, Argument::any())->shouldNotBeCalled();
+        $cartRepository->find(3)->willReturn($cart);
+        $eventDispatcher->dispatch(
+            SyliusCartEvents::CART_INITIALIZE,
+            Argument::type(GenericEvent::class)
+        )->shouldBeCalled();
+
+        $context->setCurrentCartIdentifier($cart)->shouldNotBeCalled();
 
         $this->getCart()->shouldReturn($cart);
     }
 
     function it_creates_new_cart_if_there_is_no_identifier_in_storage(
-        $context,
-        $repository,
+        CartContextInterface $context,
+        RepositoryInterface $cartRepository,
+        EventDispatcherInterface $eventDispatcher,
+        CartInterface $cart,
         FactoryInterface $cartFactory,
-        $eventDispatcher,
         CartInterface $cart
     ) {
         $context->getCurrentCartIdentifier()->willReturn(null);
         $cartFactory->createNew()->willReturn($cart);
-        $eventDispatcher->dispatch(SyliusCartEvents::CART_INITIALIZE, Argument::any())->shouldBeCalled();
+        $cartRepository->find()->shouldNotBeCalled();
+
+        $this->initialize_cart_should_be_called_on($eventDispatcher);
```

Comment:
Spec's are not the classes, so it is not a good practice to remove duplications. I found it less readable with this private method. And instead of reading what it should do, I have to jump to private declaration. And it is used only twice. Remove this method.